conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUndertowRuntime.java

```diff
@@ -103,6 +112,12 @@ public Builder paramMarker(ParamMarker value) {
             return this;
         }
 
+        @CanIgnoreReturnValue
+        public Builder deprecated(DeprecationCallback value) {
+            deprecated = Preconditions.checkNotNull(value, "deprecated is required");
+            return this;
+        }
```

Comment:
I don't think this should be configurable (much like the plain serde interface)

Reply:
the plain SerDe interface is completely implemented in conjure-undertow-lib which I disagree we should be doing for this. Plain SerDe implemented here makes sense because we do want to have tight control over the result of SerDe (else the webserver would not work correctly). On the other hand, this is similar to the markerCallbacks and I feel like clients should be free to implement this how they please.

Reply:
I think this is more similar to our exception handling code than it is to markers. While some consumers would prefer to implement their own exception handling, we don't allow it because it allows for unnecessarily complex and confusing code.
Much like exception handling, we should decide how deprecation should be handled, and do it properly for all implementations.

Reply:
In this case, we would need to pass the DefaultTaggedRegistry. Should we pass it to this builder?

Reply:
Yep, that would be great!